Jumpstart: switch from a modal to a card.

jeherve opened the pull request:
Fixes #10584 
#### Changes proposed in this Pull Request:

- Remove special jumpstart route we used to display the modal.
- Remove redirections to and from that route.
- Change the display of the Jumpstart component to a card matching the specs in #10584

![image](https://user-images.githubusercontent.com/426388/49864842-af0caa80-fe03-11e8-8fc8-2e742ce87c51.png)

#### Testing instructions:

* Start with brand new install or by using the link to reset all options at the bottom of the Jetpack menu pages.
* Visit any of the Jetpack pages and notice the new card.
* Make sure the card looks good on any device.
* Make sure the 2 actions (dismiss or jumpstart) work well.

#### Proposed changelog entry for your changes:
* Connection flow: make it clearer that Jetpack offers a set of recommended features when you start using the plugin.

jeherve committed:
Jumpstart: switch from a modal to a card.

Fixes #10584

- Remove special jumpstart route we used to display the modal.
- Remove redirections to and from that route.
- Change the display of the Jumpstart component to a card matching the specs in #10584

jeherve commented:
@keoshi Do you think you could apply some of your CSS magic to the card? I don't know that it looks good on all devices / browsers, and that everything is aligned as you'd like.

Thank you!

keoshi commented:
Will do and will also test this thoroughly in a moment!

keoshi committed:
Tweak copy strings

keoshi committed:
Update styles for mobile

keoshi committed:
Fix mobile styles

keoshi commented:
Pushed a couple of changes, should be good for a review now!

oskosk commented:
Looks like this on mobile.

![image](https://user-images.githubusercontent.com/746152/49872973-6f40c580-fdf9-11e8-912a-b4052ace0ccd.png)

keoshi commented:
> Should the "Skip and explore features" link still be present?

Good point, we should add this back below the button (and probably get rid of the cross).

jeherve committed:
Remove cross icon and use a link to offer dismissal.

@see https://github.com/Automattic/jetpack/pull/10954#issuecomment-446587855

jeherve commented:
I removed the cross and added the link back in f84ffde877c5b95824030735b67f8ff00684903a

![image](https://user-images.githubusercontent.com/426388/49874945-159fc180-fe20-11e8-9806-751da70d3fe6.png)

![image](https://user-images.githubusercontent.com/426388/49874969-26e8ce00-fe20-11e8-83b1-d1f94798b2be.png)

Maybe it should all be centered on mobile?

rickybanister commented:
I think this is a solid change. It's also a good lead-in to what an autopilot master settings toggle might be like. 

As a separate thought, how can this card and the stats placeholder below it work together in the first moment someone visits the Jetpack dashboard? Should they? What are we trying to get across in these first glances?

rickybanister commented:
My comment above is more philosophical fodder for @keoshi, nothing actionable needed at this time!

keoshi commented:
> how can this card and the stats placeholder below it work together in the first moment someone visits the Jetpack dashboard

I like where you're heading with that. Would definitely be interesting to play around with the different elements in this view to create a more compelling initial impact.

oskosk reviewed (approved):
LGTM!

keoshi commented:
@jeherve Just noticed the `Skip and explore features` link doesn't redirect users to the Settings page. Should I open a new issue for this?

jeherve commented:
> Just noticed the Skip and explore features link doesn't redirect users to the Settings page.

@keoshi I made the change in #10987.
